src/mjolnir/ingest_transit.cc

```diff
@@ -162,69 +163,60 @@ std::priority_queue<tileTransitInfo> select_transit_tiles(const boost::property_
+        if (station_in_tile != tile_info.stations.end()) {
+          tile_info.station_children.insert({{stop.parent_station, feed_path}, stop.stop_id});
         } else {
-          currTile.tile_stops.insert({stop.stop_id, feed_path});
+          // we don't have the parent station, if
+          // 1) this stop has none or 2) its parent station is in another tile
+          // TODO: need to handle the 2nd case somehow!
+          tile_info.stations.insert({stop.stop_id, feed_path});
+          tile_info.station_children.insert({{stop.stop_id, feed_path}, stop.stop_id});
         }
 
-        const gtfs::StopTimes& currStopTimes = feed.get_stop_times_for_stop(currStopId);
-
-        for (const auto& stopTime : currStopTimes) {
-          // add trip, route, agency and service_id from stop_time
-          // could use std::set to not check if it already exists
-          auto currTrip = feed.get_trip(stopTime.trip_id);
-          assert(currTrip);
-          auto currRoute = feed.get_route(currTrip->route_id);
-          assert(currRoute);
-          if (currTrip->service_id.empty()) {
-            LOG_ERROR("Missing service_id for trip");
+        for (const auto& stopTime : feed.get_stop_times_for_stop(stop.stop_id)) {
+          // add trip, route, agency and service_id from stop_time, it's the only place with that info
+          // TODO: should we throw here?
+          auto trip = feed.get_trip(stopTime.trip_id);
+          auto route = feed.get_route(trip->route_id);
+          if (!trip || !route || trip->service_id.empty()) {
+            LOG_ERROR("Missing trip or route or service_id for trip");
             continue;
           }
 
-          currTile.tile_services.insert({currTrip->service_id, feed_path});
-          currTile.tile_trips.insert({currTrip->trip_id, feed_path});
-
-          currTile.tile_shapes.insert({{currTrip->shape_id, feed_path}, currTile.tile_shapes.size()});
-          // remember the shape id such that we can access it later
-
-          currTile.tile_routes.insert(
-              {{currRoute->route_id, feed_path}, currTile.tile_routes.size()});
-          currTile.tile_agencies.insert({currRoute->agency_id, feed_path});
-          // add to the queue of tiles
+          tile_info.services.insert({trip->service_id, feed_path});
+          tile_info.trips.insert({trip->trip_id, feed_path});
+          tile_info.shapes.insert({{trip->shape_id, feed_path}, tile_info.shapes.size()});
+          tile_info.routes.insert({{route->route_id, feed_path}, tile_info.routes.size()});
+          tile_info.agencies.insert({route->agency_id, feed_path});
         }
-        tile_map[currId] = currTile;
       }
 
-      LOG_INFO("Done Sorting through " + feed_path);
+      LOG_INFO("Done parsing " + std::to_string(tile_map.size()) + " transit tiles for GTFS feed " +
+               feed_path);
     }
   }
-  std::priority_queue<tileTransitInfo> prioritized;
+  std::priority_queue<tile_transit_info_t> prioritized;
```

Comment:
while we are fixing naming:
```suggestion
  std::priority_queue<tile_transit_info_t> queue;
```